src/vector.rs

```diff
@@ -41,6 +41,48 @@ macro_rules! vec {
                 }
             }
         }
+
+        #[cfg(feature = "serde")]
+        impl<T> ::serde::Serialize for $name<T>
+            where T: ::serde::Serialize
+        {
+            fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+                where S: ::serde::Serializer
+            {
+                use ::serde::ser::SerializeSeq;
```

Comment:
why aren't we piggy-backing on the `AsRef` implementation? E.g. this function would look like `self.as_ref().serialize(serializer)`

Reply:
:heavy_check_mark: